subsys/logging/log_backend_fcb.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2020 Xylem Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log_backend.h>
+#include <logging/log_core.h>
+#include <logging/log_msg.h>
+#include <logging/log_output.h>
+#include "log_backend_std.h"
+#include <fs/fcb.h>
+#include <assert.h>
+
+static struct flash_sector sectors[CONFIG_LOG_BACKEND_FCB_SECTOR_COUNT] = {
```

Comment:
I would not configure each sector apart but rather use dedicated flash area for logging.

1. Example of sectors extraction:
``` C
static struct flash_sector log_fcb_area[CONFIG_max_sector_count + 1];
static struct fcb fcb_log_instance;

uint32_t cnt = ARRAY_SIZE(log_fcb_area);

int rc;
const struct flash_area *fap;

rc = flash_area_get_sectors(FLASH_AREA_ID(LOG_BACKEND_FCB_FLASH_AREA_LABEL), &cnt, log_fcb_area);
	return rc;
} else if (rc != 0 && rc != -ENOMEM) {
	k_panic();
}

fcb_log_instance.f_sector_cnt = cnt;
fcb_log_instance.flash_sector = log_fcb_area;
...
fcb_init()
```
2.
alternatively to this procedure flash_area_get_device() + flash_get_parameters() call might be used as FCB requires same sectors size in flash it uses. Especially if you want to have fcb sector in size of multiple physical flash sectors I would go 2..